### DISCUSSION TITLE
How to register TransientGrantingPermissionService in serenity .net 5

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6075] [Help] [answered]
In old .net framework we did,
```
registrar.RegisterInstance<IPermissionService>(
     new Serenity.Web.TransientGrantingPermissionService(new Administration.PermissionService()));
```
but in .net 5 we have 
`            services.AddSingleton<IPermissionService, Administration.PermissionService>();`

how can we register `TransientGrantingPermissionService` in .net 5

also the doc is not updated
https://github.com/serenity-is/docs/blob/a89a776a0136966926bbc5df3176d4c4dad86cc7/api/dotnet/Serenity.Net.Core/Serenity.Web/TransientGrantingPermissionService.md

**Comment by VictorTomaili** [comment, marked as answer]
You can use something like this
```cs
services.AddSingleton<IPermissionService>(services => 
    new TransientGrantingPermissionService(ActivatorUtilities.CreateInstance<PermissionService>(services)));
```

**Reply by dfaruque** [reply]
Thank you very much for the hint.
requestContext null exception occurred.
I have fixed this by the following code
```
services.AddSingleton<IPermissionService>(services =>
     new TransientGrantingPermissionService(ActivatorUtilities.CreateInstance<Administration.PermissionService>(services), 
     services.GetService<IHttpContextItemsAccessor>()));
```
